akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/scaladsl/EventConsumer.scala

```diff
@@ -0,0 +1,202 @@
+        journalPluginId: Option[String] = journalPluginId,
+        acceptedStreamIds: Set[String] = acceptedStreamIds,
+        transformationForOrigin: (String, Metadata) => Transformation = transformationForOrigin,
+        interceptor: Option[EventConsumerInterceptor] = interceptor,
+        filters: immutable.Seq[FilterCriteria] = filters): EventConsumerDestination =
+      new EventConsumerDestination(journalPluginId, acceptedStreamIds, transformationForOrigin, interceptor, filters)
+  }
+
+  object EventConsumerDestination {
+
+    /**
+     * @param acceptedStreamIds Only accept these stream ids, deny others
+     */
+    def apply(acceptedStreamIds: Set[String]): EventConsumerDestination =
+      new EventConsumerDestination(None, acceptedStreamIds, (_, _) => Transformation, None, immutable.Seq.empty)
+
+    /**
+     * @param acceptedStreamId Only accept this stream ids, deny others
+     */
+    def apply(acceptedStreamId: String): EventConsumerDestination =
+      apply(Set(acceptedStreamId))
+  }
+
+  @ApiMayChange
+  object Transformation extends Transformation {
+    // Note: this is also the empty/identity transformation
+    override private[akka] def apply(eventEnvelope: EventEnvelope[Any]): EventEnvelope[Any] =
+      eventEnvelope
+  }
+
+  private final case class MapTags(f: EventEnvelope[Any] => Set[String]) extends Transformation {
+    override private[akka] def apply(eventEnvelope: EventEnvelope[Any]): EventEnvelope[Any] = {
+      val newTags = f(eventEnvelope)
+      if (newTags eq eventEnvelope.tags) eventEnvelope
+      else
+        new EventEnvelope[Any](
+          offset = eventEnvelope.offset,
+          persistenceId = eventEnvelope.persistenceId,
+          sequenceNr = eventEnvelope.sequenceNr,
+          eventOption = eventEnvelope.eventOption,
+          timestamp = eventEnvelope.timestamp,
+          eventMetadata = eventEnvelope.eventMetadata,
+          entityType = eventEnvelope.entityType,
+          slice = eventEnvelope.slice,
+          filtered = eventEnvelope.filtered,
+          source = eventEnvelope.source,
+          tags = newTags)
+    }
+
+  }
+
+  private final case class MapPersistenceId(f: EventEnvelope[Any] => String) extends Transformation {
+    override private[akka] def apply(eventEnvelope: EventEnvelope[Any]): EventEnvelope[Any] = {
+      val newPid = f(eventEnvelope)
+      if (newPid eq eventEnvelope.persistenceId) eventEnvelope
+      else
+        new EventEnvelope[Any](
+          offset = eventEnvelope.offset,
+          persistenceId = newPid,
+          sequenceNr = eventEnvelope.sequenceNr,
+          eventOption = eventEnvelope.eventOption,
+          timestamp = eventEnvelope.timestamp,
+          eventMetadata = eventEnvelope.eventMetadata,
+          entityType = eventEnvelope.entityType,
+          slice = eventEnvelope.slice,
+          filtered = eventEnvelope.filtered,
+          source = eventEnvelope.source,
+          tags = eventEnvelope.tags)
```

Comment:
would make things more readable with a private `copyEnvelope` method in Transformation